arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -104,25 +104,33 @@ Result fromFuncMinHashMatch(char const* funcName, irs::boolean_filter* filter,
                             aql::AstNode const& args);
 
 FieldMeta::Analyzer const& FilterContext::fieldAnalyzer(
-    std::string_view name, Result& r) const noexcept {
+    std::string_view name, aql::ExpressionContext* ctx) const noexcept {
+  // FIXME(gnusi): refactor this function
+
   if (!fieldAnalyzerProvider) {
-    if (ADB_UNLIKELY(!contextAnalyzer)) {
-      TRI_ASSERT(false);
-      r = {TRI_ERROR_INTERNAL, "Malformed search/filter context"};
-    }
+    // Only possible with ArangoSearch view
     return contextAnalyzer;
   }
+
+  auto registerWarning = [ctx](std::string_view error) {
+    if (ctx) {
+      ctx->registerWarning(TRI_ERROR_BAD_PARAMETER, error);
+    }
+  };
+
   auto const& analyzer = (*fieldAnalyzerProvider)(name);
+  if (!analyzer) {
+    // Only possible with SearchAlias views
+    registerWarning(absl::StrCat("Analyzer for field '", name, "' isn't set"));
+    return FieldMeta::identity();
```

Comment:
contextAnalyzer ? contextAnalyzer : FieldMeta::identity();

Reply:
contextAnalyzer  looks more consistent in this case so +1 from me